addons/web_editor/static/src/js/widgets/widgets.js

```diff
@@ -533,22 +546,133 @@ var ImageWidget = MediaWidget.extend({
             }
 
             function _processFile(attachment, error) {
-                var $button = self.$('.o_upload_image_button');
                 if (!error) {
-                    $button.addClass('btn-success');
                     self._toggleImage(attachment, true);
                 } else {
-                    $button.addClass('btn-danger');
                     self.$el.addClass('o_has_error').find('.form-control, .custom-select').addClass('is-invalid');
                     self.$el.find('.form-text').text(error);
                 }
-
-                if (!self.multiImages) {
-                    self.trigger_up('save_request');
-                }
```

Comment:
Was this decided ? No more leaving the media dialog on upload then ?

Reply:
In the spec since the beginning, so yeah :)

Reply:
I do not see that in the specs, where are those specs ? :smile:

Reply:
"Images : The latests image uploaded or from URL appears first with a green-border" and sbu asked to me several times.
Plus, 9 days ago I asked you this :

> Do you know if it is conceivable in the _uploadImage function (in widgets.js - ImageWidget) to do something like that : Instead of closing modal when a new image is updated, we could get directly the new image next to the other ones without closing modal (and for instance having a green border around it). What do you think ?
> (branch updated)

That was about all about this.

Reply:
Yes, but I don't take decisions for functional stuff :) And I don't understand that spec like that but if SBU asked you directly, ok then :)

Reply:
He did and also explained it to me. This isn't my decision :smiley: I misspoke. I wanted your technical opinion on that one.

Reply:
My opinion is that FGI will want the opposite on the next saas version :laughing: But no problem doing this now :wink: